src/view/StrokeView.cpp

```diff
@@ -171,13 +174,15 @@ void StrokeView::drawWithPressuire()
 {
 	int count = 1;
 	double width = s->getWidth();
-	ArrayIterator<Point> points = s->pointIterator();
+    std::vector<Point>::const_iterator points = s->iteratorBegin();
 
-	Point lastPoint1 = points.next();
+    Point lastPoint1 = *points;
+    points++;
 	double dashOffset = 0;
-	while (points.hasNext())
+    while (points != s->iteratorEnd())
 	{
-		Point p = points.next();
+        Point p = *points;
+        points++;
```

Comment:
use a for loop like 
`for(auto&& p:points){` (in case of points being an container) or use
`for(auto i = s->iteratorBegin(), i != s->iteratorEnd(),++i){` in case of iterators

Reply:
added a helper template for loops with special cases for the first element,
since this was more then once the case,

https://raw.githubusercontent.com/st4ll1/xournalpp/remove_arrayiterator/src/util/LoopUtil.h

inspired by https://stackoverflow.com/questions/54831971/c-range-based-loop-with-special-case-for-first-item

i like this solution since there is no extra if in the loop